test/utils/general.spec.js

```diff
@@ -169,30 +169,30 @@ describe('services', () => {
       it('processes an array where the index field is a single value', () => {
         const inputArray = [
           { category: '1', value: 'a' },
-          { category: '1', value: 'b' }
+          { category: '1', value: 'b' },
         ];
-        expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({ '1': inputArray });
+        expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({ 1: inputArray });
       });
       it('processes an array where the index field does not exist', () => {
         const inputArray = [
           { category: '1', value: 'a' },
-          { category: '1', value: 'b' }
+          { category: '1', value: 'b' },
         ];
         expect(GeneralUtil.indexByField(inputArray, 'category2')).toStrictEqual({
-          undefined: inputArray
+          undefined: inputArray,
         });
       });
       it('processes an array where the index field can contain multiple values (array)', () => {
         const inputArray = [
           { category: ['1'], value: 'a' },
-          { category: ['1', '2'], value: 'b' }
+          { category: ['1', '2'], value: 'b' },
         ];
         expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({
-          '1': [
+          1: [
```

Comment:
It should preserve the original type here, the function does not convert the type from string to int.